src/App.jsx

```diff
@@ -1,33 +1,64 @@
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App = () => (
-  <div className="page">
-    <div className="page-content">
-      <div className="box">
-        <div className="field">
-          {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-          <label htmlFor="search-query" className="label">
-            Search movie
-          </label>
-
-          <div className="control">
-            <input
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+function getPreparedMovies(movies, { query }) {
+  let preparedMovies = [...movies];
+
+  if (query) {
+    preparedMovies = preparedMovies.filter(
+      movie => movie.title.toLowerCase().includes(query.toLowerCase().trim())
+      || movie.description.toLowerCase().includes(query.toLowerCase().trim()),
+    );
+  }
+
+  return preparedMovies;
+}
+
+export const App = () => {
+  const [query, setQuery] = useState('');
+  const visibleMovies = getPreparedMovies(
+    moviesFromServer, { query },
+  );
+
+  const filterBy = (newQuery) => {
+    setQuery(newQuery);
+  };
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <div className="box">
+          <div className="field">
+            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+            <label htmlFor="search-query" className="label">
+              Search movie
+            </label>
+
+            <div className="control">
+              <input
+                type="text"
+                id="search-query"
+                className="input"
+                placeholder="Type search word"
+                onChange={(event) => {
+                  filterBy(event.target.value);
+                }}
```

Comment:
```suggestion
                onChange={filterBy}
```
it is bad practise to make anonymous arrow function in jsx. filterBy handler can excep event and set query directly as event.target.value.